context.ts

```diff
@@ -98,4 +117,19 @@ export class Context<S extends State = Record<string, any>> {
     }
     throw err;
   }
+
+  /** Take the current request and upgrade it to a web socket, resolving with
+   * the web socket object. This will set `.respond` to `false`. */
+  async upgrade(): Promise<WebSocket> {
+    if (this.#socket) {
+      return this.#socket;
+    }
+    const { conn, r: bufReader, w: bufWriter, headers } =
+      this.request.serverRequest;
+    this.#socket = await acceptWebSocket(
```

Comment:
Not checking if the request actually is of type "Upgrade" can result in an acceptWebSocket error: https://deno.land/std@0.54.0/ws/mod.ts#L443

Acceptance check:
https://deno.land/std@0.54.0/ws/mod.ts#L397

Or was it a conscious decision, leaving it away so that the deno consumer "can/has to" implement an own check? :)

Reply:
I am not sure what you mean.  In this case if you were to do:

```ts
await context.upgrade();
```

It would reject with the error, if the request is not upgradeable (in fact there is a test for this behaviour as well).  Are you suggesting it should/could behave differently?  Like maybe add an `isUpgradable` property?

Reply:
Upon reflection, `ctx.isUpgradeable` makes sense and so I've added it.